packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -28,85 +32,295 @@ def lb_enabled():
+                'labels': {}
+            }]
+            if vip is not None:
+                app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': vip}
+            if network == 'USER':
+                app['ipAddress'] = {'networkName': 'dcos'}
+
+        app['constraints'] = [['hostname', 'CLUSTER', host]]
+        return app, uuid
+
+    def mesos_vip_app(self, num, network, host, vip, ucr=False):
+        app = None
+        uuid = None
+        port = backend_port_st + num
+        if ucr is False:
+            app, uuid = get_test_app()
+        else:
+            app, uuid = get_test_app_in_ucr()
+        app['id'] = '/viptest/' + app['id']
+        app['mem'] = 16
+        app['cpu'] = 0.01
+        app['healthChecks'] = [{
+            'protocol': 'MESOS_HTTP',
+            'path': '/ping',
+            'gracePeriodSeconds': 5,
+            'intervalSeconds': 10,
+            'timeoutSeconds': 10,
+            'maxConsecutiveFailures': 3,
+        }]
+        assert network != 'BRIDGE'
+        if network == 'USER':
+            app['ipAddress'] = {
+                'discovery': {
+                    'ports': [{
+                        'protocol': 'tcp',
+                        'name': 'test',
+                        'number': port,
+                    }]
+                }
+            }
+            app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                         '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py {}'.format(port)
+            app['ipAddress']['networkName'] = 'dcos'
+            if vip is not None:
+                app['ipAddress']['discovery']['ports'][0]['labels'] = {'VIP_0': vip}
+            app['healthChecks'][0]['port'] = port
+            app['portDefinitions'] = []
+
+        if network == 'HOST':
+            app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                         '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
+            app['portDefinitions'] = [{
+                'protocol': 'tcp',
+                'port': 0
+            }]
+            if vip is not None:
+                app['portDefinitions'][0]['labels'] = {'VIP_0': vip}
+            app['healthChecks'][0]['portIndex'] = 0
+
+        app['constraints'] = [['hostname', 'CLUSTER', host]]
+        log.info('app: {}'.format(json.dumps(app)))
+        return app, uuid
+
+    def vip_app(self, num, container, network, host, vip):
+        if container == 'UCR':
+            return self.mesos_vip_app(num, network, host, vip, ucr=True)
+        if container == 'DOCKER':
+            return self.docker_vip_app(network, host, vip)
+        if container == 'NONE':
+            return self.mesos_vip_app(num, network, host, vip, ucr=False)
+        assert False, 'unkown container option {}'.format(container)
+
+    def run(self, dcos_api_session):
+        self.log('START')
+        agents = list(dcos_api_session.all_slaves)
```

Comment:
nit: make this a copy.copy instead of list(something_that_is_already_a_list). Shows this is clearly a copy and not a cast, but that might just be my preference